stream/ResponsePartIterator.java

```diff
@@ -92,8 +92,11 @@ public boolean hasNext() {
 
     @Override
     public TransactionProto.Transaction.ResPart next() {
-        if (!hasNext()) throw new NoSuchElementException();
-        state = State.EMPTY;
-        return next;
+        if (stream.getError().isPresent()) throw TypeDBClientException.of(stream.getError().get());
+        else if (!hasNext()) throw new NoSuchElementException();
+        else {
+            state = State.EMPTY;
+            return next;
+        }
```

Comment:
we make the next() idempotently throw the same error if the user repeatedly calls it